FreeRTOS/FreeRTOS
# Add test late timer start (retry #557)

Description
-----------
This PR fixes #557 (which had to be reverted in #566) by putting a warning in TimerDemo.h about the backlog tests.  There are two distinct parts to this PR:

- Add a warning in TimerDemo.h about the backlog tests (missing in #557)
- Add another test to the existing backlog tests (same as #557)

Note that the warning in TimerDemo.h is important with or without the additional backlog test.

Test Steps
-----------
Run the timer demo in isolation (not concurrently with other demos).  Call `vTimerDemoIncludeBacklogTests( pdTRUE )` to include the new test provided in this PR.  (The test detects an issue resolved by https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/305 and identified below.)

Related Issue
-----------
https://forums.freertos.org/t/bug-xtimeristimeractive-is-returning-wrong-value-v10-3-0/9469

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
